aiokafka/abc.py

```diff
@@ -87,6 +90,17 @@ def on_partitions_assigned(self, assigned):
         pass
 
 
+class AbstractTokenProvider(BaseAbstractTokenProvider):
+    async def token(self):
```

Comment:
I think only providing a token async function with a proper specification is enough, let's remove the `run_in_executor`. By introducing the `_token` abstract function below you essentially require the developer to override it (you can't initialize an `abc` class without overriding all abstract methods), which is strange in the case where he only needs an async callback.
We can add a docstring describing how you can use synchronous callback instead.